crates/checkpoint-downloader/src/downloader.rs

```diff
@@ -558,6 +562,21 @@ fn handle_checkpoint_error(err: &RetriableClientError, next_checkpoint: u64) {
             "failed to read next checkpoint, probably not produced yet",
         );
     }
+
+    if let RetriableClientError::FallbackError(fallback_error) = err
+        && let FallbackError::RequestFailed(error) = fallback_error
```

Comment:
Nit: These two could be merged:
```suggestion
    if let RetriableClientError::FallbackError(FallbackError::RequestFailed(error)) = err
```